packages/oidc-http-server-pages/src/create-templates.ts

```diff
@@ -84,26 +84,23 @@ export function generateTemplates<TPage extends string = HttpServerPage>(
   return templates as PageTemplates<TPage>;
 }
 
-export function generateCompressedTemplates<
-  TPage extends string = HttpServerPage
->(
-  pages: Record<
-    TPage,
-    {
-      Component: Component;
-      parameters: string[];
-    }
-  >
-): void {
-  const templates = generateTemplates(pages);
-  const buffer = gzipSync(JSON.stringify(templates));
+function generateGzip(data: string): void {
+  const buffer = gzipSync(data);
   writeFileSync(path.join(__dirname, 'templates.gz'), buffer, 'binary');
+}
 
+function generateJS(data: string): void {
+  const buffer = brotliCompressSync(data, {
+    params: {
+      [zlibConstants.BROTLI_PARAM_MODE]: zlibConstants.BROTLI_MODE_TEXT,
+      [zlibConstants.BROTLI_PARAM_QUALITY]: zlibConstants.BROTLI_MAX_QUALITY,
+    },
+  });
   writeFileSync(
     path.join(__dirname, 'templates.js'),
     `
-    const { gunzipSync } = require('zlib');
-    const buffer = gunzipSync(
+    const { brotliDecompressSync } = require('zlib');
+    const buffer = brotliDecompressSync(
```

Comment:
I think ideally we'd also keep the lazy decompression that the previous code had – otherwise, loading this file would always uncompress the data blob, regardless of whether we'll actually end up using it.

(I'll probably do something similar for the devtools-connect package in general as part of the shell performance work for loading `oidc-plugin` from `devtools-connect`, it turns out instantiating the npm packages used for the HTTP server is fairly expensive, but I'm not sure about what exactly that will look like)

Reply:
I thought the conditional require in getStaticPage would do the job, is that not so?

Reply:
ah, right it will go through webpack

Reply:
Ah no right, it should address this 👍

Reply:
oh okay, I was too quick and I realised I'm not sure if conditional require even works with webpack

Reply:
@paula-stacho Yeah, the only difference with webpack is that the code will be _parsed_ regardless of whether it is used or not, but conditional _evaluation_ of modules works just fine

Reply:
yeah that makes sense. I will revert the last commits.